nixos/modules/system/etc/make-etc.sh

```diff
@@ -3,11 +3,11 @@ source $stdenv/setup
 mkdir -p $out/etc
 
 set -f
-sources_=($sources)
-targets_=($targets)
-modes_=($modes)
-users_=($users)
-groups_=($groups)
+eval "sources_=($sources)"
```

Comment:
What does the value of $sources looks like in that case?

Reply:
So this brought to my attention that $sources actually *shouldn't* be escaped, because it's a list of paths, that when escaped don't get properly imported into the store. It also means the script is passed a list of nix store paths, which are sanitized anyway.

For the others, specifically $target, assuming we have the files `foo` and `bar baz`, it would like `'foo' 'bar baz'`, whereas previously it was `foo bar baz`.

The eval is safe, as it expands to an array literal of string literals.